access-common/src/main/java/edu/unc/lib/dl/ui/service/SolrQueryLayerService.java

```diff
@@ -249,7 +247,7 @@ public SearchResultResponse performSearch(SearchRequest searchRequest) {
         // Adjust the number of results to retrieve
         if (searchState.getRowsPerPage() == null || searchState.getRowsPerPage() < 0) {
             searchState.setRowsPerPage(searchSettings.defaultPerPage);
-        } else if (searchState.getRowsPerPage() > searchSettings.getMaxPerPage()) {
+        } else if (!searchState.getAllResults() && searchState.getRowsPerPage() > searchSettings.getMaxPerPage()) {
```

Comment:
I feel a bit pedantic suggesting it, but something like `ignoreMaxRows` could be clearer about the flags purpose. To me, "getAllResults" reads as a method for retrieving a list of results, rather modifier to a query to override the row limit.